src/coreclr/vm/typehash.cpp

```diff
@@ -234,10 +229,13 @@ static DWORD HashTypeHandle(TypeHandle t)
     else if (t.IsGenericVariable())
     {
         _ASSERTE(!"Generic variables are unexpected here.");
-        retVal = t.AsTAddr();
+        // Revisit the cast to DWORD if this path becomes viable.
+        retVal = (DWORD)t.AsTAddr();
```

Comment:
It would be more appropriate to return `0` as fallback hashcode here.

Reply:
Good idea.